rust/main/chains/hyperlane-starknet/src/utils.rs

```diff
@@ -0,0 +1,305 @@
+use std::sync::Arc;
+
+use cainome::cairo_serde::CairoSerde;
+use hyperlane_core::{
+    rpc_clients::call_and_retry_n_times, ChainCommunicationError, ChainResult, HyperlaneMessage,
+    ModuleType, ReorgPeriod, TxOutcome,
+};
+use starknet::accounts::Execution;
+use starknet::{
+    accounts::SingleOwnerAccount,
+    core::{
+        chain_id::{MAINNET, SEPOLIA},
+        types::{EmittedEvent, FieldElement, MaybePendingTransactionReceipt, TransactionReceipt},
+        utils::{cairo_short_string_to_felt, CairoShortStringToFeltError},
+    },
+    providers::{jsonrpc::HttpTransport, AnyProvider, JsonRpcClient, Provider},
+    signers::LocalWallet,
+};
+use url::Url;
+
+use crate::types::{tx_receipt_to_outcome, HyH256};
+use crate::{
+    contracts::{
+        interchain_security_module::ModuleType as StarknetModuleType,
+        mailbox::Bytes as MailboxBytes, mailbox::Message,
+        validator_announce::Bytes as ValidatorAnnounceBytes,
+    },
+    HyperlaneStarknetError,
+};
+
+type TransactionReceiptResult = ChainResult<MaybePendingTransactionReceipt>;
+
+/// Polls the rpc client until the transaction receipt is available.
+pub async fn get_transaction_receipt(
+    rpc: &Arc<AnyProvider>,
+    transaction_hash: FieldElement,
+) -> TransactionReceiptResult {
+    // there is a delay between the transaction being available at the client
+    // and the sealing of the block, hence sleeping for 100ms
+    call_and_retry_n_times(
```

Comment:
btw `call_and_retry_n_times` will sleep 2 seconds between retries. Compared to the 100ms in the comment I'm not sure if it affects any latency assumptions you had. Worth either updating the comment or adding an optional `sleep_duration` parameter to `call_and_retry_n_times` that defaults to the currently used `RPC_RETRY_SLEEP_DURATION`

Reply:
good catch, added the extra parameter